### ISSUE TITLE
Bug: [ban-types] Rule does not recognize `{}` used as a value

### ISSUE THREAD

**Issue by simPod** [typescript-eslint/typescript-eslint#5186]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=JYOwLgpgTgZghgYwgAgJIB4AiyIA9IgAmAzsgEoQID2Uh6xYUoA5gDTICuIA1iFQO4gAfMgC8yAN4BfERIBQUoA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AIwEMnajTUMkRNGgB7aJHBgAviGlA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
interface I<D extends Record<string, unknown> = {}> {
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/ban-types": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

`= {}` is used as a value and it should not be reported

### Actual Result

`Don't use `{}` as a type.` even though it's not used as a type.

### Versions

See playground, all latest.
